src/core/evaluator.js

```diff
@@ -2143,80 +2160,122 @@ class PartialEvaluator {
         });
     }
 
-    function buildTextContentItem(chars) {
-      var font = textState.font;
-      var textChunk = ensureTextContentItem();
-      var width = 0;
-      var height = 0;
-      var glyphs = font.charsToGlyphs(chars);
-      for (var i = 0; i < glyphs.length; i++) {
-        var glyph = glyphs[i];
-        var glyphWidth = null;
-        if (font.vertical && glyph.vmetric) {
-          glyphWidth = glyph.vmetric[0];
+    function buildTextContentItem(chars, extraSpacing = 0) {
+      const font = textState.font;
+      let textChunk = ensureTextContentItem();
+      const glyphs = font.charsToGlyphs(chars);
+      const scale = textState.fontMatrix[0] * textState.fontSize;
+      const NormalizedUnicodes = getNormalizedUnicodes();
+      let size = 0;
+      let lastCharSize = 0;
+      let lastAdvanceSize = 0;
+      for (let i = 0, ii = glyphs.length; i < ii; i++) {
+        const glyph = glyphs[i];
+        let charSpacing =
+          textState.charSpacing + (i === ii - 1 ? extraSpacing : 0);
+
+        let glyphUnicode = glyph.unicode;
+        if (glyph.isSpace) {
+          charSpacing += textState.wordSpacing;
         } else {
-          glyphWidth = glyph.width;
+          glyphUnicode = NormalizedUnicodes[glyphUnicode] || glyphUnicode;
+          glyphUnicode = reverseIfRtl(glyphUnicode);
         }
+        textChunk.str.push(glyphUnicode);
+
+        const glyphWidth =
+          font.vertical && glyph.vmetric ? glyph.vmetric[0] : glyph.width;
 
-        var glyphUnicode = glyph.unicode;
-        var NormalizedUnicodes = getNormalizedUnicodes();
-        if (NormalizedUnicodes[glyphUnicode] !== undefined) {
-          glyphUnicode = NormalizedUnicodes[glyphUnicode];
+        let t = glyphWidth * scale;
```

Comment:
Can we, pretty please, avoid one-letter variable names for improved readability?